lisp/ess-custom.el

```diff
@@ -2896,45 +2901,87 @@ the variable `ess-help-own-frame' is non-nil."
+ ; Faces
 ;;;=====================================================
-;;; Users note: Variables with document strings starting
-;;; with a * are the ones you can generally change safely, and
-;;; may have to upon occasion.
-
-(defvar ess-function-call-face 'ess-function-call-face
-  "Face name to use for highlighting function calls.")
-
-(defvar ess-numbers-face 'ess-numbers-face
-  "Face name to use for highlighting numbers.")
-
-(defvar ess-%op%-face 'ess-%op%-face
-  "Face name to use for highlighting %op% operators.")
 
+(defconst ess-function-call-face 'ess-function-call-face)
 (defface ess-function-call-face
   '((default (:slant normal :inherit font-lock-function-name-face)))
   "Font Lock face used to highlight function calls in ess buffers."
-  :group 'ess)
+  :group 'ess-faces)
 
+(defconst ess-numbers-face 'ess-numbers-face)
 (defface ess-numbers-face
   '((default (:slant normal :inherit font-lock-type-face)))
   "Font Lock face used to highlight numbers in ess-mode buffers."
-  :group 'ess)
+  :group 'ess-faces)
 
+(defconst ess-backquoted-face 'ess-backquoted-face)
 (defface ess-backquoted-face
   '((default (:inherit default)))
   "Font Lock face for backquoted names."
-  :group 'ess)
+  :group 'ess-faces)
 
+(defconst ess-operator-face 'ess-operator-face)
 (defface ess-operator-face
   '((default (:inherit font-lock-constant-face)))
   "Font Lock face for operators."
-  :group 'ess)
+  :group 'ess-faces)
 
+(defconst ess-%op%-face 'ess-%op%-face)
 (defface ess-%op%-face
   '((default (:inherit ess-operator-face)))
   "Font Lock face used to highlight %op% operators in ess-mode buffers."
-  :group 'ess)
+  :group 'ess-faces)
+
+(defconst ess-assignment-face 'ess-assignment-face)
+(defface ess-assignment-face
+  '((default (:inherit font-lock-constant-face)))
+  "Font lock face used to highlight assignment operators."
+  :group 'ess-faces)
+
+(defconst ess-paren-face 'ess-paren-face)
+(defface ess-paren-face
+  '((default (:inherit font-lock-constant-face)))
+  "Font lock face used to highlight parentheses."
+  :group 'ess-faces)
+
+(defconst ess-operator-face 'ess-operator-face)
+(defface ess-operator-face
+  '((default (:inherit font-lock-constant-face)))
+  "Font lock face used to highlight operators."
+  :group 'ess-faces)
+
+(defconst ess-modifiers-face 'ess-modifiers-face)
+(defface ess-modifiers-face
+  '((default (:inherit font-lock-constant-face)))
+  "Font lock face used to highlight modifiers."
+  :group 'ess-faces)
+
+(defconst ess-constant-face 'ess-constant-face)
+(defface ess-constant-face
+  '((default (:inherit font-lock-type-face)))
+  "Font lock face used to highlight constants."
+  :group 'ess-faces)
+
+(defconst ess-f-t-face 'ess-f-t-face)
+(defface ess-f-t-face
+  '((default (:inherit font-lock-type-face)))
+  "Font lock face used to highlight F and T."
+  :group 'ess-faces)
+
+(defface ess-matrix-face
+  '((default (:inherit font-lock-constant-face)))
+  "Font lock face used to highlight matrices."
```

Comment:
I think this doc is imprecise. We probably highlight only row numbers. Could you please have a look?

Reply:
On my end, it highlights row and column labels when printing matrix output (that look like [1,] or [,1]) or the labels on vectors [1] when printing vector output. I've updated the doc so it's hopefully clearer.